homeassistant/components/light/__init__.py

```diff
@@ -64,72 +65,92 @@ class LightEntityFeature(IntEnum):
 ATTR_COLOR_MODE = "color_mode"
 # List of color modes supported by the light
 ATTR_SUPPORTED_COLOR_MODES = "supported_color_modes"
-# Possible color modes
-COLOR_MODE_UNKNOWN = "unknown"  # Ambiguous color mode
-COLOR_MODE_ONOFF = "onoff"  # Must be the only supported mode
-COLOR_MODE_BRIGHTNESS = "brightness"  # Must be the only supported mode
+
+
+class ColorMode(StrEnum):
+    """Possible light color modes."""
+
+    UNKNOWN = "unknown"  # Ambiguous color mode
+    ONOFF = "onoff"  # Must be the only supported mode
+    BRIGHTNESS = "brightness"  # Must be the only supported mode
+    COLOR_TEMP = "color_temp"
+    HS = "hs"
+    XY = "xy"
+    RGB = "rgb"
+    RGBW = "rgbw"
+    RGBWW = "rgbww"
+    WHITE = "white"  # Must *NOT* be the only supported mode
+
+
+# These COLOR_MODE_* constants are deprecated as of Home Assistant 2022.5.
+# Please use the LightEntityFeature enum instead.
+COLOR_MODE_UNKNOWN = "unknown"
+COLOR_MODE_ONOFF = "onoff"
+COLOR_MODE_BRIGHTNESS = "brightness"
 COLOR_MODE_COLOR_TEMP = "color_temp"
 COLOR_MODE_HS = "hs"
 COLOR_MODE_XY = "xy"
 COLOR_MODE_RGB = "rgb"
 COLOR_MODE_RGBW = "rgbw"
 COLOR_MODE_RGBWW = "rgbww"
-COLOR_MODE_WHITE = "white"  # Must *NOT* be the only supported mode
+COLOR_MODE_WHITE = "white"
 
 VALID_COLOR_MODES = {
```

Comment:
For a follow-up PR we might be able to get rid of this constant because whatever uses this constant should check if the enum has this value.